python/MooseDocs/extensions/appsyntax.py

```diff
@@ -772,13 +772,14 @@ def createMaterialize(self, parent, token, page):
         html.Tag(p, 'span', string='C++ Type:')
         html.String(p, content=cpp_type, escape=True)
 
-        doc_unit = param['doc_unit']
-        p = html.Tag(body, 'p', class_='moose-parameter-description-doc-unit')
-        html.Tag(p, 'span', string='Unit:')
-        if not doc_unit:
-            html.String(p, content='(no unit assumed)')
-        else:
-            html.String(p, content=doc_unit)
+        if cpp_type == "double" or cpp_type == "std::vector<double>":
```

Comment:
let's also keep vectors of vectors and vectors of vectors and vectors. 
and any map that ends with Real as the item or the key (terrible idea for the key)

Reply:
axctually, let's move this if condition to the "if not doc_unit" clause,
and always obey the user specification of a unit

Reply:
> always obey the user specification of a unit

I would tend to agree with this since my first draft of this capability allowed it for everything, but units don't make sense for things like booleans, (potentially) ints, etc. So we'd want to have some limits to lessen confusion on readers of the documentation.

Reply:
if the developer set a vector of bool to have a unit, then so be it

Reply:
what we could do is still show "no unit specified" on this selected list if we want. 
that s still possible while respecting the developer's unit wishes on any input

note that Real is not the only good type. 
Requesting a PostprocessorName in "m" is actually perfectly valid. It just means "hey make sure that postprocessor is in meters!"

Reply:
@GiudGiud how we have the logic in the PR right now is *too* limiting, I agree, given the example you and Kyle have named for PostprocessorNames. Keeping the message in there by default for all things might make auditors happier. But in conversations with @gambka and @bwspenc, it seems that for some items having the field visible for certain types (like bools) just ends up being confusing for users if they happen across it. 

I think our exclusion list will be a lot shorter than we think and won't change quite as much as the available types in MOOSE will overall. To avoid your scenario on `Tensor<Reals>` in a few years, perhaps we need to lead with "if not <list of types>" instead of "if <list of types>" as it is now? Then it leaves the door open to all other hypothetical types we want to add later.

Reply:
I think the right code here is this
```
        doc_unit = param['doc_unit']
        p = html.Tag(body, 'p', class_='moose-parameter-description-doc-unit')
        html.Tag(p, 'span', string='Unit:')
        if not doc_unit and cpp_type == "double" or cpp_type == "std::vector<double>":
            html.String(p, content='(no unit assumed)')
        else:
            html.String(p, content=doc_unit)
```
- developer asked for a unit: obey
- developer did not, only add the "no unit" if it s likely it could have a unit